sale_generator/models/sale_generator.py

```diff
@@ -0,0 +1,109 @@
+# coding: utf-8
+#  @author Sébastien BEAU <sebastien.beau@akretion.com>
+#  @author Abdessamad HILALI <abdessamad.hilali@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class SaleGenerator(models.Model):
+    _name = 'sale.generator'
+
+    name = fields.Char(string='Generator', default='/')
+    partner_ids = fields.Many2many(
+        comodel_name='res.partner', string="Partner")
+    sale_ids = fields.One2many(
+        comodel_name='sale.order', inverse_name='generator_id', string="Sales")
+    tmpl_sale_id = fields.Many2one(
+        comodel_name='sale.order',
+        string="Sale Template",
+        required=True,
+        domain=[('is_template', '=', True)])
+    date_order = fields.Datetime(
+        string='Date', oldname='date',
+        default=fields.datetime.now())
+    warehouse_id = fields.Many2one(
+        comodel_name='stock.warehouse',
+        required=True,
+        string="Warehouse")
+    state = fields.Selection([
+        ('draft', 'Draft'),
+        ('generating', 'Generating Order'),
+        ('done', 'Done'),
+    ], string='State', readonly=True, default='draft')
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string="Company",
+        related="warehouse_id.company_id",
+        store=True)
+
+    @api.multi
+    def _prepare_copy_vals(self, partner):
+        vals = {}
+        vals.update({
```

Comment:
`vals = {...}`
No need to initialize the variable with an empty dict just to define all values at once on the next line.